src/main/java/org/killbill/billing/plugin/adyen/dao/AdyenDao.java

```diff
@@ -264,7 +217,8 @@ public AdyenResponsesRecord addResponse(
         new WithConnectionCallback<AdyenResponsesRecord>() {
           @Override
           public AdyenResponsesRecord withConnection(final Connection conn) throws SQLException {
-            return DSL.using(conn, dialect, settings)
+            final DSLContext dslContext = DSL.using(conn, dialect, settings);
```

Comment:
⚠️ Both statements (`execute` and `fetchOne`) should be in a single database transaction, to ensure the right `record_id` is returned.

For an example:

https://github.com/killbill/killbill-stripe-plugin/blob/ec31abcfa6ef3a95a877ce05d6f9db370c17471d/src/main/java/org/killbill/billing/plugin/stripe/dao/StripeDao.java#L203